pkg/streamingpromql/engine_test.go

```diff
@@ -283,7 +475,26 @@ func TestRangeVectorSelectors(t *testing.T) {
 				defer q.Close()
 
 				res := q.Exec(context.Background())
-				require.Equal(t, expected, res)
+
+				// Because Histograms are pointers, it is hard to use Equal for the whole result
+				// Instead, compare what we can, and check values for the Histograms
+				expectedMatrix := expected.Value.(promql.Matrix)
+				resMatrix := res.Value.(promql.Matrix)
+				require.Equal(t, expectedMatrix.Len(), resMatrix.Len(), "Right number of results")
+				for i := range expectedMatrix {
+					if expectedMatrix[i].Histograms == nil {
+						require.Equal(t, expectedMatrix[i], resMatrix[i], "Results match expectation exactly (Floats)")
+					} else {
+						require.Equal(t, expectedMatrix[i].Metric, resMatrix[i].Metric, "Metric name matches")
+						require.Equal(t, expectedMatrix[i].Floats, resMatrix[i].Floats, "Float points match")
+						require.Equal(t, len(expectedMatrix[i].Histograms), len(resMatrix[i].Histograms), "Same number of histograms")
+						for j := range expectedMatrix[i].Histograms {
+							require.Equal(t, expectedMatrix[i].Histograms[j].T, resMatrix[i].Histograms[j].T, "Histogram timestamps match")
+							require.Equal(t, expectedMatrix[i].Histograms[j].H.Sum, resMatrix[i].Histograms[j].H.Sum, "Histogram Sums match")
+							require.Equal(t, expectedMatrix[i].Histograms[j].H.Count, resMatrix[i].Histograms[j].H.Count, "Histogram Counts match")
```

Comment:
Could we use `FloatHistogram.Equals()` here? Or does that not work because of float imprecision issues?

If not, should we also check the other fields on the histograms as well (eg. buckets, spans, zero count and zero threshold)?

Reply:
ask @zenador , they have the most experience with tests

Reply:
can try https://github.com/grafana/mimir/blob/main/pkg/util/test/histogram.go#L110

Reply:
Thanks!